recipes-openxt/initrdscripts/initramfs-xenclient/init.sh

```diff
@@ -27,6 +27,26 @@ ROOT_READONLY=
 DEFINIT=/sbin/init
 FIRSTBOOT_FLAG=/boot/system/firstboot
 
+is_tpm_2_0 () {
+    [ -e /sys/class/tpm/tpm0/device/description ] && cat /sys/class/tpm/tpm0/device/description | grep "2.0" &>/dev/null
```

Comment:
I don't have a 2.0 in front of me but my 1.2 does not have the description entry. You may want to consider inspecting /sys/class/tpm/tpm0/caps, grep for the TCG version line (2nd one) split on ':' and compare with second field, i.e.
`cat /sys/class/tpm/tpm0/caps | grep "TCG version" | cut -f2 -d: | tr -d " "`
For 1.2 devices that gives you the string "1.2".